apps/backend/src/groups/groups.service.ts

```diff
@@ -93,4 +112,38 @@ export class GroupsService {
 
     return groupToDelete;
   }
+
+  syncUserGroups(user: User, groups: Array<string>) {
+    // Check if user is any existing groups that they should not be
+    user.$get('groups', {include: [User]}).then((currentGroups) => {
+      currentGroups.filter(group => !groups.includes(group.name)).forEach((groupToLeave) => {
+        this.removeUserFromGroup(groupToLeave, user);
```

Comment:
shouldn't this be awaited to confirm that the user is removed from the group? especially since there's a potential for an error to be thrown